mlflow/gateway/base_models.py

```diff
@@ -0,0 +1,36 @@
+from pydantic import BaseModel, Extra
+
+
+class RequestModel(
+    BaseModel,
+    # Allow extra fields for pydantic request models, e.g. to support
+    # vendor-specific embeddings parameters
+    extra=Extra.allow,
+):
+    """
+    A pydantic model representing Gateway request data, such as a chat or completions request
+    """
+
+
+class ResponseModel(
+    BaseModel,
+    # Ignore extra fields for pydantic response models to ensure a consistent response
+    # experience for clients across different backends
+    extra=Extra.ignore,
+):
+    """
+    A pydantic model representing Gateway response data, such as information about a Gateway
+    Route returned in response to a GetRoute request
+    """
+
+
+class ConfigModel(
+    BaseModel,
+    # Allow extra fields for pydantic config models, e.g. to support
+    # vendor-specific configuration attributes
+    extra=Extra.allow,
```

Comment:
Do we need `extra=Extra.allow` here?

Reply:
I don't think there's harm in allowing extra fields for configs right now

Reply:
^ Updated the comment.

Reply:
Absolutely, makes sense. Changed to `ignore`